### ISSUE TITLE
feature request: spectra dynamic when the region moves

### ISSUE THREAD

**Issue by teuben** [CARTAvis/carta#30]
When the cursor moves, the X Y and Z slices dynamically change right away, this is very important for data browsing. But to increase signal to noise, a region can be better. But when I select a region, the spectrum is only updated after I've dropped the region in another location.
Again, sorry for the comparison,but in ds9 the spectrum dynamically changed when the regions is moved, before you drop it again. This is a very useful feature. So if it's possible to do this....

**Comment by kswang1029** [comment]
A quick response about why we require a drop action to trigger profile update is for the consideration of large cubes with lots of channels. With small cubes, calculating region spectral profile is quick (enough) to catch up the speed of cursor movement. This is not the case for large(r) cubes. What would happen in ds9 when there are lots of channels?🤔

**Comment by teuben** [comment]
yes, on ds9 it will slow you down when the region gets large, But perhaps
like your recomput button somewhere else in carta, this could be an option
to make it dynamic. I think most people will use smallish regions though,
it took me years of ds9 usage until i ran into the slowness for large
regions.

**Comment by veggiesaurus** [comment]
> When the cursor moves, the X Y and Z slices dynamically change right away, this is very important for data browsing. But to increase signal to noise, a region can be better. But when I select a region, the spectrum is only updated after I've dropped the region in another location.
> Again, sorry for the comparison,but in ds9 the spectrum dynamically changed when the regions is moved, before you drop it again. This is a very useful feature. So if it's possible to do this....

I think we may be able to handle this appropriately, now that we have region spectral profile cancellation working nicely. The problem is that for large files, this will be prohibitively slow, and will degrade performance in general. So we may need a region size threshold for this :thinking:

**Comment by teuben** [comment]
A fancy option might be (if you can program it) to threshold this on a preset delta-time, e.g. 500ms, once you detect that updates are not optimal.  Users could then set their tolerance lower or higher. I would imagine if it's over the threshold, you would only update once the region was dropped again.
